### PR TITLE
[FLINK-22576] Set status to 'Open' when unassigning

### PR DESCRIPTION
**Pull request by zentol**

### CODE REVIEW

File: flink_jira_rule.py [hunk 1]
```diff
@@ -100,6 +100,7 @@ def close_issue(self, key):
     def unassign(self, key):
         if not self.is_dry_run:
             self.jira_client.assign_issue(key, None)
+            self.jira_client.set_issue_status(key, "Open")
```
Comment (tillrohrmann) [inline flink_jira_rule.py]: Does this work even if the JIRA bot is not assigned? I think I always need to assign myself to set a ticket back into "Open".

**Review by tillrohrmann — COMMENTED** [review]
Thanks for creating this PR @zentol. I had a question whether the JIRA bot has more permissions than we have so that it can set a ticket back into "Open" w/o being assigned to it.

File: flink_jira_rule.py (hunk 1 above)
Comment (zentol) [reply flink_jira_rule.py]: I suppose you're right; we first need to assign the bot, then stop progress, and then unassign the ticket.

**Comment by zentol** [comment]
@tillrohrmann ping

**Review by tillrohrmann — APPROVED** [review]
LGTM. +1 for merging.

**Pull request merged**